### PR TITLE
Drop Pulpcore 3.16 - 3.18, add EL 9 support

### PR DESCRIPTION
**Pull request by ekohl**
While it doesn't modify any code, does reduce the testing matrix. It's also chosen because 3.21 and 3.22 have EL9 and that keeps the tesing matrix easier.

Currently untested, going to rely on CI.

### CODE REVIEW

**Comment by evgeni** [comment]
```
  /bin/sh: line 1: hostname: command not found
```

seems we need a hack here (again?)

File: manifests/repo.pp [hunk 1]
```diff
@@ -20,13 +20,16 @@
...
 
-  package { 'pulpcore-dnf-module':
-    ensure      => $dist_tag,
-    name        => 'pulpcore',
-    enable_only => true,
-    provider    => 'dnfmodule',
-    require     => File['/etc/yum.repos.d/pulpcore.repo'],
-    notify      => Anchor['pulpcore::repo'],
+  # Only EL8 has DNF modules
+  if $dist_tag == 'el8' {
```
Comment (evgeni) [inline manifests/repo.pp:24]: the tests still expect this on el9…

Comment (ekohl) [reply manifests/repo.pp:24]: Yes, I thought about fixing the other issues while I was at it but that it was filled with rabbits just to get an EL9 box within Vagrant.

File: spec/classes/repo_spec.rb [hunk 2]
```diff
@@ -13,14 +13,18 @@
...
 
-      it 'configures the pulpcore module' do
-        is_expected.to contain_package('pulpcore-dnf-module')
-          .with_name('pulpcore')
-          .with_ensure(/^el\d+/)
-          .with_enable_only(true)
-          .with_provider('dnfmodule')
-          .that_requires('File[/etc/yum.repos.d/pulpcore.repo]')
-          .that_notifies('Anchor[pulpcore::repo]')
+      if os_facts[:op]['release']['major'] == '8'
```
Comment (evgeni) [inline spec/classes/repo_spec.rb:16]:
```suggestion
      if os_facts[:os]['release']['major'] == '8'
```

;)

Comment (ekohl) [reply spec/classes/repo_spec.rb:16]: I'll just leave this here until the other parts have been sorted.

**Comment by ekohl** [comment]
This is now green, except on Puppet 6.

**Review by ehelms — APPROVED** [review]

**Pull request merged**